sdk/src/azure/iot/az_iot_provisioning_client.c

```diff
@@ -276,6 +277,35 @@ AZ_INLINE az_result _az_iot_provisioning_client_parse_payload_error_code(
   return AZ_ERROR_ITEM_NOT_FOUND;
 }
 
+AZ_INLINE az_result
+_az_iot_provisioning_client_get_json_object_span(az_json_reader* jr, az_span* out_span)
+{
+  *out_span = jr->token.slice;
```

Comment:
ugh, this is so generic there seams like a helper func somewhere?

Reply:
@ericwol-msft maybe I'm not interpreting this correctly: you mean in the JSON parser?

@ahsonkhan do you have anything like this available that I missed or was added after the feature branch was created?

Reply:
I am assuming you are trying to get a span that contains the entire sub-object from within the JSON.

So, for example, in the following:
```json
{ "sub-object": { "name": "Cris", "complex-object": {}, "numberOfEmbeddedDevices": 25}}
```
You want to extract out:
```json
{ "name": "Cris", "complex-object": {}, "numberOfEmbeddedDevices": 25}
```

Correct?

What you have is the way to go about it, building on top of the provided methods. There isn't a built-in API to extract multi-token sub-objects out.

We could expose the internal property like `bytes_consumed` to indicate at what offset/index the token started, but that doesn't help too much here, since you'd still have to do the following to get the length of the sub-object and create a span: `end_object_index - start_object_index  + 1`.

Reply:
Thanks @ahsonkhan . Since extracting sub-objects is something fairly common, it may be useful to add this functionality to the JSon parser.